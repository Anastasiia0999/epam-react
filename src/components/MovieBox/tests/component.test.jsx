import React from 'react';
import { mount, shallow } from 'enzyme';

import {MovieBox} from "../index";

describe('<MovieBox />', () =>{
    let wrapper;
    let instance;

    const props = {
        openedMovie:{
            "id": 1,
            "title": "The Shawshank Redemptoin",
            "posterUrl": "https://upload.wikimedia.org/wikipedia/ru/thumb/d/de/Movie_poster_the_shawshank_redemption.jpg/240px-Movie_poster_the_shawshank_redemption.jpg",
            "stars": 2,
            "likes": 22,
            "genres": [
                "Crime",
                "Drama"
            ],
            "actorsIds": [
                1,
                2
            ],
            "director": "Frank Darabont",
            "description": "Chronicles the experiences of "
        },
        actors:[{
            "id": 1,
            "name": "Morgan Freeman",
            "imgUrl": "https://st.kp.yandex.net/images/actor_iphone/iphone360_6750.jpg",
            "biography": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
        },
        {
            "id": 2,
            "name": "Bob Gunton",
            "imgUrl": "https://i.pinimg.com/736x/9a/69/1e/9a691e7471b7cff69a86262bbcbb2e4b--bob-gunton-epic-film.jpg",
            "biography": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
        }],
        loadingMovie:false,
        loadingActors: false,
        fields:{
            "director": "Director",
            "actors": "Actors",
            "genres": "Genres",
            "description": "Description",
            "likes-name": "Likes",
            "edit-btn": "Edit",
            "delete-btn": "Delete"
        },
        deleteMovieDispatch: jest.fn(),
        getMovieDispatch:  jest.fn(),
        getActorsDispatch:  jest.fn()
    };

    beforeEach(() => {
        wrapper = shallow(<MovieBox {...props} />);
        instance = wrapper.instance();
    });

    it('should render correctly', () =>{
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('correct movie`s title render', () =>{
        expect(props.openedMovie.title).toBe("The Shawshank Redemptoin");
    });

    it('correct movie`s poster link render', () =>{
        expect(props.openedMovie.posterUrl).toBe("https://upload.wikimedia.org/wikipedia/ru/thumb/d/de/Movie_poster_the_shawshank_redemption.jpg/240px-Movie_poster_the_shawshank_redemption.jpg");
    });

    it('correct director render', () =>{
        expect(props.openedMovie.director).toBe("Frank Darabont");
    });

    it('correct description render', () =>{
        expect(props.openedMovie.description).toBe("Chronicles the experiences of ");
    });

    it('correct likes number render', () =>{
        expect(props.openedMovie.likes).toBe("22");
    });

    it('correct stars number render', () =>{
        expect(props.openedMovie.likes).toBe("2");
    });


});


