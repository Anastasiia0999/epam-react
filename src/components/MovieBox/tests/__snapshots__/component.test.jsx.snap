// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieBox /> should render correctly 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"card card-box col-12\\"><div class=\\"card-header movie-box_header my-3\\"><h2>The Shawshank Redemptoin</h2></div><div class=\\"card-body movie-box col-12\\"><div class=\\"movie-box_info\\"><div class=\\"movie-box_img\\"><img src=\\"https://upload.wikimedia.org/wikipedia/ru/thumb/d/de/Movie_poster_the_shawshank_redemption.jpg/240px-Movie_poster_the_shawshank_redemption.jpg\\" class=\\"rounded movie-box_image\\" alt=\\"poster\\"/></div><div class=\\"movie-details ml-3\\"><h5><b>Director:</b> Frank Darabont</h5><div><b>Actors: </b> <h4><a class=\\"link-actor\\" href=\\"/actor/1\\">Morgan Freeman</a></h4><h4><a class=\\"link-actor\\" href=\\"/actor/2\\">Bob Gunton</a></h4></div><h5><b>Genres: </b> Crime, Drama</h5><hr/><h5 class=\\"likes\\">Likes 22</h5><hr/></div></div><p class=\\"description\\" data-spy=\\"scroll\\"><b>Description:</b> Chronicles the experiences of </p></div><div class=\\"buttons_movie-box\\"><button class=\\"btn btn-info m-3\\">Edit</button><button class=\\"btn btn-danger m-3\\">Delete</button></div></div></div></div>"`;
